packages/grid/x-data-grid/src/hooks/features/rows/gridRowsUtils.ts

```diff
@@ -23,6 +33,82 @@ export function checkGridRowIdIsValid(
   }
 }
 
+export const getRowIdFromRowModel = (
+  rowModel: GridRowModel,
+  getRowId?: GridRowIdGetter,
+  detailErrorMessage?: string,
+): GridRowId => {
+  const id = getRowId ? getRowId(rowModel) : rowModel.id;
+  checkGridRowIdIsValid(id, rowModel, detailErrorMessage);
+  return id;
+};
+
+export const createRowsInternalCache = ({
+  rows,
+  getRowId,
+  loadingProp,
+}: {
+  getRowId: DataGridProcessedProps['getRowId'];
+  loadingProp: DataGridProcessedProps['loading'];
+  rows: GridRowsProp;
```

Comment:
What do you think of making explicit that they are props? `Pick` could also be used.

```suggestion
  getRowIdProp: DataGridProcessedProps['getRowId'];
  loadingProp: DataGridProcessedProps['loading'];
  rowsProp: DataGridProcessedProps['rows'];
```

Reply:
`rows` can come from `setRows` so it would not be 100% accurate
But maybe that's not a real issue though

Reply:
Done